pkg/cloud/azure/actuators/machine/actuator.go

```diff
@@ -128,72 +147,78 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 	if err != nil {
 		return errors.Errorf("failed to create scope: %+v", err)
 	}
-
 	defer scope.Close()
 
-	computeSvc := compute.NewService(scope.Scope)
-	networkSvc := network.NewService(scope.Scope)
-
 	bootstrapToken, err := a.checkControlPlaneMachines(scope, cluster, machine)
 	if err != nil {
 		return errors.Wrapf(err, "failed to check control plane machines in cluster %s", cluster.Name)
 	}
 
-	kubeConfig, err := a.GetKubeConfig(cluster, nil)
-	if err != nil {
-		return errors.Wrapf(err, "failed to retrieve kubeconfig while creating machine %q", machine.Name)
+	// kubeConfig, err := a.GetKubeConfig(cluster, nil)
+	// if err != nil {
+	// 	return errors.Wrapf(err, "failed to retrieve kubeconfig while creating machine %q", machine.Name)
+	// }
+
+	networkInterfaceSpec := networkinterfaces.Spec{
```

Comment:
consider a factory method to return the networkinterfaces.Spec.

e.g.
```
func makeNICSpec(role NodeRole) *networkinterfaces.Spec
```

Reply:
Next PR please, currently its already too much of a change

Reply:
Fine with punting that to next PR.